.github/workflows/cicd.yaml

```diff
@@ -0,0 +1,208 @@
+# CI/CD workflow for the groundlight-sdk-python repository. We lint, test, deploy docs, and publish
+# to pypi.
+name: cicd
+on: [push]
+
+env:
+  PYTHON_VERSION: "3.10"
+  POETRY_VERSION: "1.4.0"
+jobs:
+  # Run our linter on every push to the repository.
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: get code
+        uses: actions/checkout@v3
+      - name: install python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: install poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: ${{ env.POETRY_VERSION }}
+      - name: show python version ${{ env.PYTHON_VERSION }}
+        run: |
+          poetry run python --version
+      - name: install linter dependencies
+        run: |
+          make install-lint
+      - name: lint
+        run: |
+          make lint
+
+  # Run integration tests against the API. For efficiency, we only run one version of python on
+  # non-main branches.
+  test-simple:
+    runs-on: ubuntu-latest
+    steps:
+      - name: get code
+        uses: actions/checkout@v3
+      - name: install python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: install poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: ${{ env.POETRY_VERSION }}
+      - name: show python version ${{ env.PYTHON_VERSION }}
+        run: |
+          poetry run python --version
+      - name: install dependencies (without extras)
+        run: make install
+      # TODO: Should we run all tests against the prod API?
+      - name: run tests
+        env:
+          # This is associated with the "sdk-integ-test" user, credentials on 1password
+          GROUNDLIGHT_API_TOKEN: ${{ secrets.GROUNDLIGHT_API_TOKEN }}
+        run: make test-integ
```

Comment:
Note: these tests are currently failing because of an issue with `get_detector_by_name()` -- I'll fix that in a different PR.

Reply:
Fixed and merged!